plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.sdk;
+
+import org.eclipse.osgi.service.debug.DebugOptions;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+
+public class CloudSdkManager {
+
+  private static final String OPTION_MANAGED_CLOUD_SDK =
+      "com.google.cloud.tools.eclipse.sdk/managed.cloud.sdk";
+
+  public static boolean isManaged() {
+    BundleContext context = FrameworkUtil.getBundle(CloudSdkManager.class).getBundleContext();
+    DebugOptions debugOptions = context.getService(context.getServiceReference(DebugOptions.class));
+    return debugOptions.getBooleanOption(OPTION_MANAGED_CLOUD_SDK, false);
```

Comment:
It is possible for `getService()` to return `null`.

Reply:
Really in practice? Anyway, I'll check null.